crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -67,17 +67,20 @@ def transition
     state = params[:state] # State of node transitioning
     name = params[:name] # Name of node transitioning
 
-    status, response = @service_object.transition(id, name, state)
-
-    if status != 200
-      render :text => response, :status => status
+    unless NodeObject::API_REACHABLE_STATES.include?(state)
+      render :text => "State #{state} is invalid", :status => 400
```

Comment:
perhaps some quoting 

"State '#{state}' is not a valid state"

just in case there are some strange whitespace issues..